runners/spark/job-server/spark_job_server.gradle

```diff
@@ -52,9 +52,9 @@ configurations.all {
 dependencies {
   implementation project(sparkRunnerProject)
   permitUnusedDeclared project(sparkRunnerProject)
-  implementation project(path: sparkRunnerProject, configuration: "compileOnly")
+  implementation project(sparkRunnerProject)
   validatesPortableRunner project(path: sparkRunnerProject, configuration: "testRuntimeMigration")
-  validatesPortableRunner project(path: sparkRunnerProject, configuration: "compileOnly")
```

Comment:
Same question here, and in spark_runner

Reply:
The implementation change may affect this artifact- things that are listed as compileOnly deps of sparkRunnerProject will not be listed as <provided> of the job-server artifact- they will only exist as <provided> transitively.

Reply:
@ibzib do you think this is ok?

Reply:
Actually, I'll revise this. previously, this wouldn't have listed transitive deps as provided, it would just have declared a dependency on the library itself, just a different configuration of it (compileOnly vs the full target), so shouldn't affect the output artifact

Reply:
I think this caused an issue: https://issues.apache.org/jira/browse/BEAM-13569